### ISSUE TITLE
AC Alarm on GPS103 Protocol

### ISSUE THREAD

**Issue by jlms77** [vitalidze/traccar-web#972]
Hi,

I've been testing traccar for a few days with this web gui and works very well.

I'm trying to get the alarm signals for the Coban GPS which uses the GPS103 protocol, i manage to get the SOS alarm from the database in the table positions using the "others" field, i can get the following value:

{"alarm":"sos","ip":"x.x.x.x","distance":0.0,"totalDistance":337.17}

The problem i have is that the "ac alarm" event is not going to the database, i know that the server is receiving the signal because i manage to identify this event in the log:

2016-11-29 16:45:00 DEBUG: [8884492D: 5001 < x.x.x.x] HEX: 696d65693a3836323130363032313233373731362c616320616c61726d2c313631313239313634352c2c462c3230343435372e3030302c412c313031302e323738332c4e2c30363434312e303237342c572c302e30302c2c3b
2016-11-29 16:45:00  INFO: [8884492D] id: 862106021237716, time: 2016-11-29 16:44:57, lat: 10.17131, lon: -64.68379, speed: 0.0, course: 0.0

When i convert this in the HEX converter i get:

imei:862106021237716,ac alarm,1611291645,,F,204457.000,A,1010.2783,N,06441.0274,W,0.00,,;

This particular event does not  get to the database, is there any recommendation to get it working?

Thanks for your help.

**Comment by vitalidze** [comment]
Currently such events are not supported by this project. It requires some changes in the event posting system.

Actually I thought about re-using similar functionality from "backend" project. There is a possibility to write events to the database. For now I still haven't made up my mind on how to tie up events from "backend" with events from "Traccar Web UI Mod".

**Comment by jlms77** [comment]
Thanks for your fast response! 

I thought it was an easy solution, this alarm comes in the same format than the position signals and sos signal

Positions: imei:359710044145648,tracker,1611300928,,F,134318.000,A,1010.2956,N,06441.0425,W,0.00,0;

SOS Alarm: imei:359710044145648,sos,1611300928,,F,134318.000,A,1010.2956,N,06441.0425,W,0.00,0;

AC Alarm: imei:862106021237716,ac alarm,1611291645,,F,204457.000,A,1010.2783,N,06441.0274,W,0.00,,;

Basically is the same message and your system is capable to write them in the database, can you point me in the right direction to tell me where to check if we can add more alarms for this particular GPS?

Maybe the same routine that writes the SOS Alarm to the database can work or something.

**Comment by vitalidze** [comment]
> Basically is the same message and your system is capable to write them in the database, can you point me in the right direction to tell me where to check if we can add more alarms for this particular GPS?

I think you need to check code of the protocol implementation in "backend" project: https://github.com/tananaev/traccar/tree/master/src/org/traccar/protocol. This project is a "View" of the data collected by the backend.

**Comment by jlms77** [comment]
Hi vitalidze,

In fact the alarm is already added to the code, i posted the same question in traccar forum and got the response very fast just like you, thanks for your help!

**Comment by vitalidze** [comment]
Great, I am happy that it works for you!
